Modules/Components/itkImageRegistrationMethodv4/include/selxItkImageRegistrationMethodv4Component.hxx

```diff
@@ -396,6 +446,55 @@ ItkImageRegistrationMethodv4Component< Dimensionality, TPixel, InternalComputati
     // Smooth by specified gaussian sigmas for each level.  These values are specified in
     // physical units.
     this->m_ImageRegistrationMethodv4Filter->SetSmoothingSigmasPerLevel( smoothingSigmasPerLevel );
+  } else if( criterion.first == "RescaleIntensity" ) {
+    if( criterion.second.size() != 2 ) {
+      this->m_Logger.Log(LogLevel::ERR, "Expected two values for RescaleIntensity (min, max), got {0}.", criterion.second.size());
+      meetsCriteria = false;
+    }
+    else {
+      this->m_RescaleIntensity = criterion.second;
+      meetsCriteria = true;
+    }
+  } else if( criterion.first == "InvertIntensity" ) {
+    if( criterion.second.size() != 1 ) {
+      this->m_Logger.Log(LogLevel::ERR, "Expected one value for InvertIntensity (True or False), got {0}.", criterion.second.size());
+      meetsCriteria = false;
+    } else {
+      if( criterion.second[0] == "True" ) {
```

Comment:
Looks like we could use a helper function here, to convert "True" or "False" to bool. Because I see similar code in selxItkGradientDescentOptimizerv4Component.hxx and selxItkSyNImageRegistrationMethodComponent.hxx  Anyway, that could be for later  :-)

Reply:
Yes a lot the criterion code could be cleaned up with a few helper functions

Reply:
OK, criteria handling helper functions would be nice, but for another pull request `:-)`